UnderlyingOptionsPS/FTPS3-Compare.ps1

```diff
@@ -23,6 +17,83 @@ class InfoClass {
 	[string]$friendlyName #For Text message header
 }
 
+function GetFilesListAsArray($user,$pass,$local_target,$ftp_uri,$subfolder){
+ # ftp address from where to download the files
+ [system.URI] $uri = $ftp_uri + $subfolder
+ 
+$ftp=[system.net.ftpwebrequest]::Create($uri)
+ 
+if($user)
+ {
+	$ftp.Credentials=New-Object System.Net.NetworkCredential($user,$pass)
+ }
+ #Get a list of files in the current directory.
+ #Use ListDirectoryDetails instead if you need date, size and other additional file information.
+ $ftp.Method=[system.net.WebRequestMethods+ftp]::ListDirectoryDetails
+ $ftp.UsePassive=$true
+ 
+try
+{
+	 $response=$ftp.GetResponse()
+	 $strm=$response.GetResponseStream()
+	 $reader=New-Object System.IO.StreamReader($strm,'UTF-8')
+	 $list=$reader.ReadToEnd()
+	 $lines=$list.Split("`n")
+	 return $lines
+}
+ catch{
+	$_|fl * -Force
+ }
+}
+
+#################################################    Downloads Delta   ################################################################################
+
+function DownloadFile ($sourceuri,$targetpath,$username,$password){
+ # Create a FTPWebRequest object to handle the connection to the ftp server
+ $ftprequest = [System.Net.FtpWebRequest]::create($sourceuri)
+ 
+# set the request's network credentials for an authenticated connection
+ $ftprequest.Credentials = New-Object System.Net.NetworkCredential($username,$password)
+ 
+$ftprequest.Method = [System.Net.WebRequestMethods+Ftp]::DownloadFile
+ $ftprequest.UseBinary = $true
+ $ftprequest.KeepAlive = $false
+ 
+# send the ftp request to the server
+ $ftpresponse = $ftprequest.GetResponse()
+ 
+# get a download stream from the server response
+ $responsestream = $ftpresponse.GetResponseStream()
+ 
+# create the target file on the local system and the download buffer
+ try
+ {
+ $targetfile = New-Object IO.FileStream ($targetpath,[IO.FileMode]::Create)
+ "File created: $targetpath"
+ [byte[]]$readbuffer = New-Object byte[] 1024
+ 
+# loop through the download stream and send the data to the target file
+ do{
+ $readlength = $responsestream.Read($readbuffer,0,1024)
+ $targetfile.Write($readbuffer,0,$readlength)
+ }
+ while ($readlength -ne 0)
+ 
+$targetfile.close()
+ }
+ catch
+ {
+ $_|fl * -Force
+ }
+ 
+}
+
+#Set permissions and import necessary libraries
+import-module "C:\Program Files (x86)\AWS Tools\PowerShell\AWSPowerShell\AWSPowerShell.psd1"
+Set-AWSCredential -ProfileName S3access
+Initialize-AWSDefaults
+
+
```

Comment:
I always define the functions up front, it makes it easier to follow the flow when you're down in the execution